privval/file.go

```diff
@@ -184,8 +183,12 @@ func NewFilePV(privKey crypto.PrivKey, keyFilePath, stateFilePath string) *FileP
 
 // GenFilePV generates a new validator with randomly generated private key
 // and sets the filePaths, but does not call Save().
-func GenFilePV(keyFilePath, stateFilePath string) *FilePV {
-	return NewFilePV(ed25519.GenPrivKey(), keyFilePath, stateFilePath)
+func GenFilePV(keyFilePath, stateFilePath string, keyGen func() (crypto.PrivKey, error)) *FilePV {
+	key, err := keyGen()
```

Comment:
Do we really need this to return an error? Again, changes in all previous files are only needed because we have this new requirement.

Reply:
errors are always better than panics imo. since comet is a library the caller should decide how to handle the error not the library